brave-grpc/src/test/java/com/github/kristofa/brave/grpc/BraveGrpcInterceptorsTest.java

```diff
@@ -0,0 +1,173 @@
+
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.ServerInterceptors;
+import io.grpc.Status;
+import io.grpc.examples.helloworld.GreeterGrpc;
+import io.grpc.examples.helloworld.GreeterGrpc.GreeterBlockingStub;
+import io.grpc.examples.helloworld.GreeterGrpc.GreeterFutureStub;
+import io.grpc.examples.helloworld.HelloReply;
+import io.grpc.examples.helloworld.HelloRequest;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+public class BraveGrpcInterceptorsTest {
+
+    private static final HelloRequest HELLO_REQUEST = HelloRequest.newBuilder()
+        .setName("brave")
+        .build();
+
+    private Server server;
+    private ManagedChannel channel;
+    private Brave brave;
+    private boolean enableSampling;
+
+    @Before
+    public void before() throws Exception {
+        enableSampling = true;
+        SpanCollectorForTesting.getInstance().getCollectedSpans().clear();
+
+        final Brave.Builder builder = new Brave.Builder();
+        brave = builder
+            .spanCollector(SpanCollectorForTesting.getInstance())
+            .traceSampler(new ExplicitSampler())
+            .build();
+
+        int serverPort = pickUnusedPort();
+        server = ServerBuilder.forPort(serverPort)
+            .addService(ServerInterceptors.intercept(
+                GreeterGrpc.bindService(new GreeterImpl()), new BraveGrpcServerInterceptor(brave)))
+            .build()
+            .start();
+
+        channel = ManagedChannelBuilder.forAddress("localhost", serverPort)
+            .intercept(new BraveGrpcClientInterceptor("helloWorldCaller", brave))
+            .usePlaintext(true)
+            .build();
+    }
+
+    @Test
+    public void testBlockingUnaryCall() throws Exception {
+        GreeterBlockingStub stub = GreeterGrpc.newBlockingStub(channel);
+        HelloReply reply = stub.sayHello(HELLO_REQUEST);
+        assertThat(reply.getMessage(), is("Hello brave"));
+        validateSpans();
+    }
+
+    @Test
+    public void testAsyncUnaryCall() throws Exception {
+        GreeterFutureStub futureStub = GreeterGrpc.newFutureStub(channel);
+        ListenableFuture<HelloReply> helloReplyListenableFuture = futureStub.sayHello(HELLO_REQUEST);
+        HelloReply reply = helloReplyListenableFuture.get();
+        assertThat(reply.getMessage(), is("Hello brave"));
+        validateSpans();
+    }
+
+    @Test
+    public void usesExistingTraceId() throws Exception {
+        LocalTracer localTracer = brave.localTracer();
+        SpanId spanId = localTracer.startNewSpan("localSpan", "myop");
+        GreeterBlockingStub stub = GreeterGrpc.newBlockingStub(channel);
+        HelloReply reply = stub.sayHello(HELLO_REQUEST);
```

Comment:
oh wait.. we are testing parent id, here. I think you should create a SpanId that has parent id set.